src/Components/CreateImageWizardV2/steps/Oscap/Oscap.tsx

```diff
@@ -98,6 +103,25 @@ const ProfileSelector = () => {
     }
   }, [data]);
 
+  useEffect(() => {
```

Comment:
This is this components 3rd useEffect... it is getting a bit hard to keep track of all the side effects...

I'm also a bit worried about a possible async bug. I'm not sure it is a valid concern, but could it ever happen that this useEffect doesn't get triggered because data?.packages doesn't change in time? For instance, if the Next button somehow gets clicked very quickly after changing the profile and the request is still in flight?

Could we run this code inside the `handleSelect()` function instead? We might need to create a small helper function to keep the code clean. Maybe we could make it async and somehow ensure that data.packages is correct, perhaps by waiting if isFetching is true or something?

Reply:
Fair point. I've previously tried to handle dispatching within the `handleSelect` function. 

What I got stuck on is the asynchronicity of store update - when handling things within `handleSelect` (or a helper function called from it) the store stays one step behind the desired state - when you select a profile, no packages get added. When you change the profile, only packages for the first profile get added.

The only solution I've found so far was the `useEffect`. :thinking: